src/ophys_etl/modules/roi_cell_classifier/compute_classifier_artifacts.py

```diff
@@ -0,0 +1,275 @@
+        avg_img = normalize_array(array=avg_img,
+                                  lower_cutoff=q0,
+                                  upper_cutoff=q1)
+
+        q0, q1 = np.quantile(corr_img, quantiles)
+        corr_img = normalize_array(array=corr_img,
+                                   lower_cutoff=q0,
+                                   upper_cutoff=q1)
+
+        self.logger.info("Normalized images...")
+
+        with open(roi_path, "rb") as in_file:
+            extract_roi_list = sanitize_extract_roi_list(
+                json.load(in_file))
+
+        self.logger.info("Creating and writing ROI artifacts...")
+        for roi in extract_roi_list:
+            self.write_thumbnails(extract_roi=roi,
+                                  max_img=max_img,
+                                  avg_img=avg_img,
+                                  corr_img=corr_img,
+                                  exp_id=exp_id,
+                                  out_dir=out_dir)
+
+        self.logger.info(f"Created ROI artifacts in {time.time()-t0:.0f} "
+                         "seconds.")
+
+    def get_max_avg(self,
+                    video_path: pathlib.Path) -> Dict[str, np.ndarray]:
+        """Compute the mean and max projection from a movie on disk.
+
+        Parameters
+        ----------
+        video_path : pathlib.Path
+            Path to HDF5 file containing full movie. Movie is assume to be
+            stored in a dataset named "data".
+
+        Returns
+        -------
+        projections : Dict[str, np.ndarray]
+            Dictionary of image projects of the movie.
+
+            "mean" : Mean projection of the movie.
+            "max" : Max projection of the movie.
+        """
+        with h5py.File(video_path, "r") as in_file:
+            video_data = in_file["data"][()]
+        max_img = np.max(video_data, axis=0)
+        avg_img = np.mean(video_data, axis=0)
+
+        return {"max": max_img, "avg": avg_img}
+
+    def write_thumbnails(self,
+                         extract_roi: ExtractROI,
+                         max_img: np.ndarray,
+                         avg_img: np.ndarray,
+                         corr_img: np.ndarray,
+                         out_dir: pathlib.Path,
+                         exp_id: int):
+        """Compute image cutout artifacts for an ROI.
+
+        Parameters
+        ----------
+        extract_roi : ophys_etl.types.ExtractROI
+            ROI containing bounding box size and location.
+        max_img : np.ndarray
+            Max projection of movie.
+        avg_img : np.ndarray
+            Mean projection of movie.
+        corr_img : np.ndarray
+            Correlation projection of movie.
+        out_dir : pathlib.Path
+            Output path to write png files to.
+        exp_id : int
+            Id of experiment where these ROIs and images come from.
+        """
+        desired_shape = (self.args['cutout_size'], self.args['cutout_size'])
```

Comment:
There is too much logic in this method. It should be modularized into several functions

Reply:
Moved to their own methods.